### PR TITLE
Feat issue #98: merging jobs and bootstrap section

### LINKED ISSUE #98: Merge jobs top level section with bootstrap section
Following this comment from @rdettai : https://github.com/quickwit-oss/helm-charts/pull/95#issuecomment-2249826979

We're opening this issue in order to move the `jobs` top level section content (`extraVolumes`, `extraVolumeMounts` and `initContainers`) into the `bootstrap` section.

It'll be a breaking change, so to version and document in the README.

### PR DESCRIPTION
**Pull request by idrissneumann**
Fix #98

### CODE REVIEW

**Comment by idrissneumann** [comment]
Test helm template with this value file:

```yaml
---
config:
  default_index_root_uri: s3://qw-indexes
  storage:
    s3:
      endpoint: https://s3.fr-par.scw.cloud
      region: fr-par
      access_key_id: SCWXXXXXX
      secret_access_key: XXXXXXXXX

environment:
  QW_METASTORE_URI: s3://qw-indexes

# Seed configuration
seed:
  indexes:
    - version: 0.8
      index_id: my-index
      doc_mapping:
        field_mappings:
          - name: timestamp
            type: datetime
            fast: true
            input_formats:
              - unix_timestamp
            output_format: unix_timestamp_secs
          - name: body
            type: text
        timestamp_field: timestamp
      search_settings:
        default_search_fields: [body]
      indexing_settings:
        merge_policy:
          type: limit_merge
          max_merge_ops: 3
          merge_factor: 10
          max_merge_factor: 12

  sources: []

bootstrap:
  enabled: true
  sources:
    initContainers: &initContainers
      - name: hello
        image: busybox:1.28
        command: ['echo', 'heyyyyy']
  indexes:
    initContainers: *initContainers
```

Result:

```shell
$ helm template . --values values.yaml --values values2.yaml --namespace=quickwit|grep -i initContainers -A5
      initContainers:
        - command:
          - echo
          - heyyyyy
          image: busybox:1.28
          name: hello
```

**Comment by idrissneumann** [comment]
Tests on kind with the same value file:

```shell
$ helm template . --values values.yaml --values values2.yaml --namespace=quickwit|kubectl -n quickwit apply -f -
$ kubectl -n quickwit logs release-name-quickwit-index-my-index-8npw8 -c hello
heyyyyy
$ kubectl -n quickwit port-forward release-name-quickwit-searcher-0 7280:7280
```

=> Init container has been successfully run ✅
=> The index has been successfully created on quickwit ✅

Screenshot 2024-08-01 at 11 17 09

File: charts/quickwit/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: quickwit
 description: Sub-second search & analytics engine on cloud storage.
 type: application
-version: 0.6.5
+version: 0.7.0
 appVersion: "v0.8.1"
```
Comment (fmassot) [inline charts/quickwit/Chart.yaml:6]: can you update the app version to 0.8.2 too?

**Review by fmassot — APPROVED** [review]

File: charts/quickwit/Chart.yaml (hunk 1 above)
Comment (idrissneumann) [reply charts/quickwit/Chart.yaml:6]: Yeah sure

Comment (idrissneumann) [reply charts/quickwit/Chart.yaml]: Done

**Pull request merged**